lib/cloudregister/registerutils.py

```diff
@@ -482,15 +482,16 @@ def get_credentials(credentials_file):
     password = None
     if not os.path.exists(credentials_file):
         return (username, password)
-    credentials = open(credentials_file).readlines()
-    for entry in credentials:
-        if entry.startswith('username'):
-            username = entry.split('=')[-1].strip()
-        elif entry.startswith('password'):
-            password = entry.split('=')[-1].strip()
-        else:
-            logging.warning('Found unknown entry in '
-                            'credentials file "%s"' % entry)
+    with open(credentials_file) as cred_file:
+        credentials = cred_file.readlines()
+        for entry in credentials:
```

Comment:
All the logic doing work on the file data can be outside of the context manager. This allows the file to close right after readlines.